- implementar consistencia de entrada
- sobrescrever valores do grafo?

caminho
-> run -> handle_communication -> get_outgoing_links_expected_rewards

estrut: data

{'gneE33': []}
{'gneE32': [], 'gneE45-n': [], 'gneE63-n': []}
{'gneE34': [], 'gneE58': []}
{'gneE35': [], 'gneE56': [], 'gneE64-n': []}
{'gneE36': []}
{'gneE37': [], 'gneE60': []}
{'gneE38': [], 'gneE49': [], 'gneE66-n': []}
{'gneE39': []}
{'gneE40': [], 'gneE61': []}
{'gneE41': [], 'gneE53-n': [], 'gneE65-n': []}
{'gneE42': []}
{'gneE44': [], 'gneE51': []}
{'gneE45': [], 'gneE47-n': [], 'gneE52': []}
{'gneE47': [], 'gneE48-n': [], 'gneE54': [], 'gneE55-n': []}
{'gneE48': [], 'gneE49-n': [], 'gneE59': []}
{'gneE53': [], 'gneE54-n': [], 'gneE62': []}
{'gneE55': [], 'gneE56-n': [], 'gneE57': []}
{'gneE63': []}
{'gneE64': []}
{'gneE65': []}
{'gneE66': []}

erro2 (3 obj):
Traceback (most recent call last):
  File "/home/hugobbi/UFRGS/bolsa/grafos_programas/SUMO-QL-GRAPH/simulations/sumo_run.py", line 251, in handle_learning
    agents[vehicle_id].learn(action, origin_node, destination_node, reward)
  File "/home/hugobbi/UFRGS/bolsa/grafos_programas/SUMO-QL-GRAPH/sumo_ql/agent/q_learning.py", line 120, in learn
    (reward - self.__avg_rewards[current_state][action]) / self.__visits[current_state][action]
ValueError: operands could not be broadcast together with shapes (3,) (2,) 

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/hugobbi/UFRGS/bolsa/grafos_programas/SUMO-QL-GRAPH/simulations/sumo_run.py", line 380, in <module>
    main()
  File "/home/hugobbi/UFRGS/bolsa/grafos_programas/SUMO-QL-GRAPH/simulations/sumo_run.py", line 376, in main
    run_sim(options)
  File "/home/hugobbi/UFRGS/bolsa/grafos_programas/SUMO-QL-GRAPH/simulations/sumo_run.py", line 289, in run_sim
    run(iteration)
  File "/home/hugobbi/UFRGS/bolsa/grafos_programas/SUMO-QL-GRAPH/simulations/sumo_run.py", line 186, in run
    handle_communication(vehicle_id, vehicle['current_state'])
  File "/home/hugobbi/UFRGS/bolsa/grafos_programas/SUMO-QL-GRAPH/simulations/sumo_run.py", line 284, in handle_communication
    handle_learning(vehicle_id, origin, destination, expected_reward)
  File "/home/hugobbi/UFRGS/bolsa/grafos_programas/SUMO-QL-GRAPH/simulations/sumo_run.py", line 257, in handle_learning
    raise Exception(exception).with_traceback(exception.__traceback__)
  File "/home/hugobbi/UFRGS/bolsa/grafos_programas/SUMO-QL-GRAPH/simulations/sumo_run.py", line 251, in handle_learning
    agents[vehicle_id].learn(action, origin_node, destination_node, reward)
  File "/home/hugobbi/UFRGS/bolsa/grafos_programas/SUMO-QL-GRAPH/sumo_ql/agent/q_learning.py", line 120, in learn
    (reward - self.__avg_rewards[current_state][action]) / self.__visits[current_state][action]
Exception: operands could not be broadcast together with shapes (3,) (2,) 
Error: tcpip::Socket::recvAndCheck @ recv: peer shutdown7ms, vehicles TOT 4504 ACT 767 BUF
Quitting (on error).
malloc_consolidate(): unaligned fastbin chunk detected
